assets/javascripts/modules/checkout/promoCode.js

```diff
@@ -22,24 +23,79 @@ define([
         bindExtraKeyListener.alreadyBound = true;
     }
 
-    function displayError(message) {
-        message = message || 'Invalid promo code, please try again.';
+    function unapplyPromotionToRatePlans() {
+        $ratePlanFields.each(function(el) {
+            var $el = $(el),
+                currency = $el.data('currency'),
+                $label = $('#label-for-' + $el.val() + '-' + currency),
+                newDisplayPrice = $el.data('option-mirror-label-default');
+
+            $label.html(newDisplayPrice);
+            $el.attr('data-option-mirror-label', newDisplayPrice);
+            if ($el.attr('checked')) {
+                bean.fire(el, 'change');
+            }
+        });
+    }
+
+    function formatMoney(currencyIdentifier, amount) {
+        // rounds any remainder up tp 1p so we always quote a higher price than the user might eventually get charged.
+        var price2dp = Math.ceil(amount * 100) / 100,
+            priceNoDot00 = price2dp.toFixed(2).replace(/\.00$/, '');
+
+        return currencyIdentifier + priceNoDot00;
+    }
+
+    function applyPromotionToRatePlans(promotion) {
+        if (promotion.promotionType.amount > 0) {
```

Comment:
I appreciate there's probably a deadline on this so this doesn't block merging but I think it'd be nice to implement this price panel thing as a client side template with some kind of model that gets modified and redisplayed when a promotion is applied

So we have say something like

```javascript
plans: [
    { id: 1, price: 11.99, name: 'Monthly' },
    { id: 2, price: 30.99, name: 'Quarterly' },
    { id: 3, price: 120.99, name: 'Yearly' }
]
```
which get fed into a template rendered with the lodash template function (or react if you're feeling ambitious)
Then instead of all of this HTML stuff we can do something nice like 
```javascript
function applyPromotionToRatePlans(promotion) {
    var newPlans = map(plans, function(plan) {
        return { 
            name: plan.name, 
            price: plan.price * 1 + (discountPercent / 100),  //etc
            id: plan.id 
        }
    });
    $sidebar.innerHTML = template('plans.html', newPlans)
}
```

I find JS like this where the logic is mixed in with DOM operations quite difficult to read!
Alex and I got lodash templates working on membership which was a bit of a battle but I think its worth it